### PR TITLE
fix(siblings) Combine siblings data but remove duplicate data

### PR DESCRIPTION
**Pull request by chriscollins3456**
We were seeing duplicate tags and terms if the two siblings had the same tag or term before they were merged as siblings. This uses `unionBy` from lodash to combine arrays of objects but removes duplicates based on a field. So you can see I remove duplicates based on the `urn` for the the fields we're seeing duplicates for.

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-web-react/src/app/entity/shared/siblingUtils.ts [hunk 1]
```diff
@@ -41,6 +42,31 @@ const combineMerge = (target, source, options) => {
...
+const mergeTags = (destinationArray, sourceArray, _options) => {
+    return unionBy(destinationArray, sourceArray, 'tag.urn');
+};
+
+const mergeTerms = (destinationArray, sourceArray, _options) => {
+    return unionBy(destinationArray, sourceArray, 'term.urn');
+};
+
+const mergeAssertions = (destinationArray, sourceArray, _options) => {
+    return unionBy(destinationArray, sourceArray, 'urn');
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/siblingUtils.ts:54]: We don't need assertion.urn or something here, right?

File: datahub-web-react/src/app/entity/shared/siblingUtils.ts [hunk 2]
```diff
@@ -51,6 +77,7 @@ const customMerge = (isPrimary, key) => {
     if (key === 'tags' || key === 'terms' || key === 'assertions') {
         return (secondary, primary) => {
             return merge(secondary, primary, {
+                arrayMerge: getArrayMergeFunction(key),
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/siblingUtils.ts:80]: Nice!

File: datahub-web-react/src/app/entity/shared/siblingUtils.ts [hunk 3]
```diff
@@ -41,6 +42,31 @@ const combineMerge = (target, source, options) => {
...
+function getArrayMergeFunction(key) {
+    switch (key) {
+        case 'tags':
+            return mergeTags;
+        case 'terms':
+            return mergeTerms;
+        case 'assertions':
+            return mergeAssertions;
+        default:
+            return undefined;
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/siblingUtils.ts:66]: Will this cause any runtime issues when we add new arrays to our objects?

**Review by jjoyce0510 — COMMENTED** [review]
A couple questions, looks good overall

File: datahub-web-react/src/app/entity/shared/siblingUtils.ts (hunk 3 above)
Comment (chriscollins3456) [reply datahub-web-react/src/app/entity/shared/siblingUtils.ts:66]: no sir - originally we weren't even setting `arrayMerge` which effectively we were setting `arrayMerge: undefined`. So it doesn't break anything, just defaults to how we were doing it before this fix. If we add more arrays to our objects that we need to remove duplicates to, we will need to add another custom array merge function like we have for these 3 fields so far.

**Review by jjoyce0510 — APPROVED** [review]

**Pull request merged**